src/Core/Silk.NET.Core/Miscellaneous/BreakneckLock.cs

```diff
@@ -0,0 +1,197 @@
+﻿// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+// Sourced from https://github.com/john-h-k/SpinLockSlim under the MIT license
+
+// MIT License
+// 
+// Copyright (c) 2019 John Kelly
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+// ReSharper disable RedundantAssignment
+
+namespace Silk.NET.Core
+{
+    /// <summary>
+    /// Provided a lightweight spin lock for synchronization in high performance
+    /// scenarios with a low hold time
+    /// </summary>
+    /// <remarks>
+    /// This lock is very performant, but it is very dangerous (hence breakneck).
+    /// It's recommended to use the framework-provided locks where possible.
+    /// </remarks>
+    public struct BreakneckLock
```

Comment:
Why is this in core?

Reply:
Where else would it be? It may be used elsewhere in the codebase in the future, and is not specific to windowing.

Reply:
SDL - if it is used anywhere else, I'd just copy paste it. Sure bit of extra work, but much better then having a public type `BreakneckLock`

Reply:
Personally very against this. Copying and pasting code across codebases is not the way to go when you can have a common assembly - this is literally why the core exists.

Besides, it will probably be useful for performance junkies using Silk.NET and will be documented at some point in the future.

Reply:
Fair point, though in that case I'd name it something different - BreakneckLock just sounds like "this should be internal but we couldn't be bothered to make it"

Reply:
Hmm yeah, I still want a word that conveys both the speed and lack of safety in this lock though and breakneck is all I could come up with.

Reply:
`UnsafeFastLock`?

Reply:
Ideally one word, not keen on those it doesn't really roll off the tongue well.

Reply:
That's the point though?

Reply:
I don't want people to not use this type, I just want to convey that it's quite easy to go wrong with it which is fine as all of Silk.NET is quite easy to go wrong with.

Reply:
I'd argue people that want to use this should use https://github.com/john-h-k/SpinLockSlim instead

Reply:
I mean there is zero difference between the two, and the benefit of this being in core is it's free - no need to take on a dependency. It will pretty much stay 1:1 with the original as part of our monthly updates.